modules/network/src/integrationTest/java/org/apache/ignite/internal/network/netty/ItConnectionManagerTest.java

```diff
@@ -295,6 +299,95 @@ public void testStopTwice() throws Exception {
         server.close();
     }
 
+    /**
+     * Tests that if two nodes are opening channels to each other, only one channel survives.
+     *
+     * @throws Exception If failed.
+     */
+    @RepeatedTest(100)
+    public void testOneChannelLeftIfConnectToEachOther() throws Exception {
+        try (
+                ConnectionManagerWrapper manager1 = startManager(4000);
+                ConnectionManagerWrapper manager2 = startManager(4001)
+        ) {
+            CompletableFuture<NettySender> fut1 = manager1.openChannelTo(manager2).toCompletableFuture();
+            CompletableFuture<NettySender> fut2 = manager2.openChannelTo(manager1).toCompletableFuture();
+
+            NettySender sender1 = null;
+            NettySender sender2 = null;
+
+            try {
+                sender1 = fut1.get(1, TimeUnit.SECONDS);
+            } catch (Exception ignored) {
+                // No-op.
+            }
+            try {
+                sender2 = fut2.get(1, TimeUnit.SECONDS);
+            } catch (Exception ignored) {
+                // No-op.
+            }
+
+            NettySender highlander = null;
+
+            assertTrue(sender1 != null || sender2 != null);
+
+            if (sender1 != null && sender1.isOpen()) {
+                highlander = sender1;
+
+                boolean sender2NullOrClosed = sender2 == null || !sender2.isOpen();
+
+                assertTrue(sender2NullOrClosed);
+            }
+
+            if (sender2 != null && sender2.isOpen()) {
+                highlander = sender2;
+
+                boolean sender1NullOrClosed = sender1 == null || !sender1.isOpen();
+
+                assertTrue(sender1NullOrClosed);
+            }
+
+            assertNotNull(highlander);
+            assertTrue(highlander.isOpen());
+
+            assertTrue(
+                    waitForCondition(
+                            () -> manager1.channels().size() == 1 && manager2.channels().size() == 1,
+                            TimeUnit.SECONDS.toMillis(1)
+                    )
+            );
+
+            CompletableFuture<NettySender> channelFut1 = manager1.connectionManager.channel(
+                    manager2.connectionManager.consistentId(),
+                    ChannelType.DEFAULT,
+                    manager2.connectionManager.localAddress()
+            ).toCompletableFuture();
+
+            CompletableFuture<NettySender> channelFut2 = manager2.connectionManager.channel(
+                    manager1.connectionManager.consistentId(),
+                    ChannelType.DEFAULT,
+                    manager1.connectionManager.localAddress()
+            ).toCompletableFuture();
+
+            assertThat(channelFut1, futureCompleted());
```

Comment:
I should go like "assert that <object> <verb> <noun>" or something.
`assertThat(channelFut, is(completedFuture()))` would look more natural. Current version just doesn't seem right. May I ask you to rename it?

Reply:
what is `is`?